docs/content/docs/connectors/datastream/pulsar.md

```diff
@@ -400,6 +405,264 @@ We can not promise consistency in this scenario.
+The default delayer is never delay the message dispatching. Using `MessageDelayer.fixed(Duration)` for
+making all the messages be delayed in a fixed duration. You can also implement the `MessageDelayer`
+interface for making messages be dispatched in different time.
+
+{{< hint warning >}}
+The dispatch time should be calculated by using `PulsarSinkContext.processTime()`.
+{{< /hint >}}
+
+### Configurable Options
+
+You can set arbitrary options for `PulsarClient`, `PulsarAdmin`, Pulsar `Producer` and `PulsarSink`
+by using `setConfig(ConfigOption<T>, T)`, `setConfig(Configuration)` and `setConfig(Properties)`.
+
+#### PulsarClient and PulsarAdmin Options
+
+Just read the [PulsarClient options](#pulsarclient-options) and [PulsarAdmin options](#pulsaradmin-options) defined in source part.
+
+#### Pulsar Producer Options
+
+Pulsar connector uses the Producer API to send messages. It extracts most parts of
+Pulsar's `ProducerConfigurationData` as Flink configuration options in `PulsarSinkOptions`.
+
+{{< generated/pulsar_producer_configuration >}}
+
+#### PulsarSink Options
+
+The configuration options below are mainly used for customizing the performance and message
+sending behavior. You can just leave them alone if you do not meet any performance issues.
+
+{{< generated/pulsar_sink_configuration >}}
+
+### Brief Design Rationale
+
+Pulsar sink follow the Sink API defined in 
+[FLIP-191](https://cwiki.apache.org/confluence/display/FLINK/FLIP-191%3A+Extend+unified+Sink+interface+to+support+small+file+compaction).
+
+#### Stateless SinkWriter
+
+In `EXACTLY_ONCE` mode, Pulsar sink does not store transaction information in checkpoint, 
+which means after a restart, a new bunch of transaction will be created.
+Any message in previous pending transactions will either be aborted or timeout 
+(anyway they are not visible to the downstream Pulsar consumer). 
+Pulsar team is working to optimize the resource cost bought by unfinished pending transaction.
+
+#### Pulsar Schema Evolution
+
+[Pulsar Schema Evolution](https://pulsar.apache.org/docs/en/schema-evolution-compatibility/) allows
+users to reuse the same Flink job after certain "allowed" data model changes, like adding or deleting
+a field in a AVRO-based Pojo class. Please note that users can specify Pulsar schema validation rules
+and define auto schema update. For more details please refer to the doc above.
+
+## Known Issues
+
+### No TransactionCoordinatorNotFound, but automatic reconnect
+
+Pulsar's transaction is still in active development and not stable enough. The new Pulsar 2.9.2
+introduce [a break change](https://github.com/apache/pulsar/pull/13135) in transaction comparing its previous version.
+Any Pulsar users who use 2.9.2 or above Pulsar server with an older Pulsar client would meet a `TransactionCoordinatorNotFound` exception.
+
+You should keep using the latest `pulsar-client-all` for resolving this issue.
+
```

Comment:
I remember there was a paragraph for the EXACTLY_ONCE semantics can bring low throughput in small volume scenario, do we not keep it ?

Reply:
I don't think it's an issue.

Reply:
I think it's a expected behaviour, but I feel it's worth mentioning since we've seen similar complaints from source, how about let's put it elsewhere ?